src/librustc/middle/const_eval.rs

```diff
@@ -1092,11 +1108,71 @@ pub fn eval_const_expr_partial<'tcx>(tcx: &ty::ctxt<'tcx>,
       hir::ExprBlock(ref block) => {
         match block.expr {
             Some(ref expr) => try!(eval_const_expr_partial(tcx, &**expr, ty_hint, fn_args)),
-            None => Int(0)
+            None => unreachable!(),
```

Comment:
this is unrelated, but I was not able to trigger this code-path, so I marked it as unreachable